### PR TITLE
Add `cst`, a simplified version of `container-structure-test`

### PR DESCRIPTION
**Pull request by errordeveloper**
This is a starting point for #5, to close the issue we actually need to add some tests to the most critical images.

### CODE REVIEW

File: images/cst/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# syntax=docker/dockerfile:1.1-experimental
+
+# Copyright 2020 Authors of Cilium
+# SPDX-License-Identifier: Apache-2.0
+
+ARG GOLANG_IMAGE=golang:1.14.3@sha256:1e36f8e9ac49d5ee6d72e969382a698614551a59f4533d5d61590e3deeb543a7
```
Comment (tklauser) [inline images/cst/Dockerfile:6]: Any reason not to use `golang:1.14.4` here?

Comment (errordeveloper) [reply images/cst/Dockerfile:6]: It's just not been updated yet, as this same image is used in a few other places, I'd rather not update the Go version in this PR.

Comment (tklauser) [reply images/cst/Dockerfile:6]: Ah, didn't realize that this was copied from upstream. LGTM in that case.

**Comment by nebril** [comment]
LGTM overall, there seems to be some problem with the check still though.

**Comment by errordeveloper** [comment]
@nebril thanks! the linter doesn't recognise new dockerfile features, will have to ignore those for now :)

File: images/cst/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+FROM --platform=linux/amd64 ${GOLANG_IMAGE} as go-builder
+
+COPY . /go/src/github.com/cilium/image-tools/images/cst
+RUN mkdir -p /out/linux/amd64/bin /out/linux/arm64/bin
+
+WORKDIR /go/src/github.com/cilium/image-tools/images/cst 
+
+# hadolint ignore=SC2215
+RUN --mount=target=/root/.cache,type=cache --mount=target=/go/pkg/mod,type=cache \
```
Comment (errordeveloper) [inline images/cst/Dockerfile:17]: @nebril so the linter was tripping over this, it was interpreting this as if user made a typo basically...

**Pull request merged**